include/caffe/net.hpp

```diff
@@ -192,7 +192,8 @@ class Net {
   /// @brief Append a new bottom blob to the net.
   int AppendBottom(const NetParameter& param, const int layer_id,
                    const int bottom_id, set<string>* available_blobs,
-                   map<string, int>* blob_name_to_idx);
+                   map<string, int>* blob_name_to_idx,
+                   bool skip_propagate = false);
```

Comment:
Ok, I'll move the param retrieval into `AppendBottom` and I'll leave the check into `Net::Init`